### PR TITLE
Fix symlink crashes (alternative to #1842)

### LINKED ISSUE #1227: Crash when accessing alias folders in Browser
When using the browser room to navigate to a file outside of the stuff folder (ie. Downloads or etc.), inside of My Computer are 2 paths. The first allows you to eventually navigate to the desired folder. If you try to access Macintosh HD, inside of Volumes (the 2nd path), there is an immediate crash. 
screen shot 2017-05-25 at 8 57 33 pm

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/45585007-crash-when-accessing-alias-folders-in-browser?utm_campaign=plugin&utm_content=tracker%2F33713530&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F33713530&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by shun-iwasawa**
This fixes #1227 and #984 , with keeping the symbolic link available.
This is alternative to #1842.

Please note that OT can handle symbolic links (which can be created with `ln -s` command), but seems to have no capability to resolve macOS aliases (created by `File > Make Alias` command, or by pressing `Command-L`) due to the Qt bug ([QTBUG-23927](https://bugreports.qt.io/browse/QTBUG-23927)).
Actually I wasn't aware of the difference of them before investigating this.

### CODE REVIEW

**Comment by shun-iwasawa** [comment]
Jenkins

**Comment by shun-iwasawa** [comment]
@artisteacher 
The crash seemed to be due to recursive call of the function `FileBrowser::setFolder()`, happened when the symbolic link in the browser was clicked.
Can you please check if this PR would resolve the issue?

**Comment by artisteacher** [comment]
@shun-iwasawa this works perfectly to fix the symlink folder crash #1227 but does not resolve the symlink file crash #984.

**Comment by shun-iwasawa** [comment]
@artisteacher Thank you for testing. I'll look into it.

**Comment by shun-iwasawa** [comment]
I've tested #984 and here are the results ;

### Test1 - Loading GIF alias
1. Drag & drop GIF file from the Stuff folder to the desktop. (As you pointed, the file was not moved but its alias was created)
1. Select the desktop folder and load the file (= the alias file with the same name as original) into OT.

**Result**: The file failed to be loaded properly. OT did not crash.

### Test2 - Loading TIF alias
1. Drag & drop TIF file from the Stuff folder to the desktop. (Same as Test1, alias was created)
1. Select the desktop folder and load the file into OT.

**Result**: Same as Test1. The file failed to be loaded properly. OT did not crash. 

### Test3 - Rendering files into the same folder as the alias
1. Keeping the GIF & TIF aliases in the desktop, render sequential TIF files into the desktop.  

**Result**: The files are rendered with no problem. OT did not crash.

So, to recap,  I couldn't reproduce the crash mentioned in #984 . The alias which is created via drag&drop the file from the stuff folder to desktop seems to be the same as the ones which can be created by `File > Make Alias` command. However, since it has the same extension as the original, OT misunderstood that they are supported file format.
( Actually, until checking this I didn't know that you need both `ffmpeg` and `ffprobe` to be kept in the same folder in order to use ffmpeg file I/O. )

@artisteacher Can you please let me know what I'm overlooking in order to reproduce the crash?

**Comment by shun-iwasawa** [comment]
Ah! I could reproduce the crash. I missed to have ffprobe on Test1 yesterday.
I believe the crash occurs when the ffmpeg tried to open a file with unexpected content.
I'll try to fix this, probably adding some validation check.

**Comment by shun-iwasawa** [comment]
Added an exception handling. Now OT just stops loading without crash for alias files.

Jenkins again

**Comment by artisteacher** [comment]
This fixes the gif problem - and also most of #1876! However I can still get a crash with a jpg alias and a .txt file renamed as a jpg.
[jpg_crash.txt](https://github.com/opentoonz/opentoonz/files/1958148/jpg_crash.txt)

**Comment by shun-iwasawa** [comment]
Jenkins again

**Comment by masafumi-inoue** [comment]
LGTM
